recipe_scrapers/__init__.py

```diff
@@ -605,22 +607,40 @@ def scraper_exists_for(url_path: str) -> bool:
     return host_name in get_supported_urls()
 
 
+def check_scraper_allowed(url_path, host_name, **options: Any) -> bool:
+    """Checks the website's robots.txt file to see if scraping is allowed.
+    If scraping is allowed, returns true. Note that this does not check to see
+    when the last time robots.txt was checked, so this could lead to constant pings
+    on robots.txt"""
+    if not options.get("robots.txt", True):
+        return True
+    else:
+        url = urljoin(host_name, 'robots.txt')
+        rp = urllib.robotparser.RobotFileParser()
+        rp.set_url(url)
+        rp.read()
```

Comment:
Hi @dragonpop76 - thanks for contributing this.

I have a memory from using `urllib.robotparser` in the past that many sites block its' requests (including to `robots.txt`!) because it uses the default Python user-agent.

I'll try to re-verify that behaviour soon, but in the meantime could you investigate whether we could use the existing `requests` dependency to retrieve the file, and then use the [`robotparser.parse(lines)` method](https://docs.python.org/3/library/urllib.robotparser.html?highlight=robotparser#urllib.robotparser.RobotFileParser.parse) to interpret the results?

That way, we're using the same HTTP client for both `robots.txt` retrieval and page content retrieval, making for a less confusing situation overall.

Reply:
I'll look into making that adjustment, will update the pr with the changes